x/ibc/04-channel/keeper/handshake.go

```diff
@@ -0,0 +1,368 @@
+package keeper
+
+import (
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ics03types "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
+	ics23 "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment"
+)
+
+// ChanOpenInit is called by a module to initiate a channel opening handshake with
+// a module on another chain.
+func (k Keeper) ChanOpenInit(
+	ctx sdk.Context,
+	order types.ChannelOrder,
+	connectionHops []string,
+	portID,
+	channelID string,
+	counterparty types.Counterparty,
+	version string,
+) (string, error) {
+	// TODO: abortTransactionUnless(validateChannelIdentifier(portIdentifier, channelIdentifier))
+	if len(connectionHops) != 1 {
+		return "", types.ErrInvalidConnectionHops(k.codespace)
+	}
+
+	_, found := k.GetChannel(ctx, portID, channelID)
+	if found {
+		return "", types.ErrChannelExists(k.codespace)
+	}
+
+	connection, found := k.connectionKeeper.GetConnection(ctx, connectionHops[0])
+	if !found {
+		return "", ics03types.ErrConnectionNotFound(k.codespace)
+	}
+
+	// TODO: inconsistency on ICS03 (`none`) and ICS04 (`CLOSED`)
```

Comment:
Is this inconsistency cleared up?

Reply:
There should be no inconsistency here, `NONE` should not be stored as a state.